### DISCUSSION TITLE
Resubscribe on a error

### DISCUSSION THREAD

**Discussion by madrigall** [enisdenjo/graphql-ws#295] [Q&A] [answered]
Hello! I need to always keep the subscription active, if an error comes from the server I would like to handle it, but the socket closes immediately without going to the handler. What is the way to recreate the connection so that the subscription resumes after the error?  I am using apollo. Thank you for your reply.

**Comment by enisdenjo** [comment, marked as answer]
If the server closed the socket, there's nothing the client can do to keep the socket open, the server has to handle the errors.

However, if you're talking about actual instances of `Error` being thrown causing the client to never reconnect, [`isFatalConnectionProblem`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.ClientOptions.md#isfatalconnectionproblem) is something you should look at.
